config_parser.cc

```diff
@@ -203,13 +205,22 @@ bool NginxConfigParser::Parse(std::istream* config_file, NginxConfig* config) {
         // Error.
         break;
       }
+      if (bracket_parity <= 0) {
+          // Error. (this close bracket has no matching open)
+          break;
+      }
+      bracket_parity--;
       config_stack.pop();
     } else if (token_type == TOKEN_TYPE_EOF) {
       if (last_token_type != TOKEN_TYPE_STATEMENT_END &&
           last_token_type != TOKEN_TYPE_END_BLOCK) {
         // Error.
         break;
       }
+      if (bracket_parity != 0) {
```

Comment:
I guess you can add this if-statement to the previous one to shrink the number of line changes but might make the if-statement more complex... this looks simple enough. Its up to you.